pkg/monitor/event.go

```diff
@@ -259,7 +267,22 @@ func recordAddOrUpdateEvent(
 			condition.Locator = fmt.Sprintf("%s hmsg/%s", condition.Locator, hashStr)
 		}
 
-		fmt.Printf("processed event: %+v\nresulting new interval: %s from: %s to %s\n", *obj, message, pathoFrom, to)
+		// For our debug message, capture/print only the fields we're interested in.
+		summaryObj := struct {
+			Name      string
+			Namespace string
+			Reason    string
+			Message   string
+			Count     int32
+		}{
+			Name:      obj.InvolvedObject.Name,
+			Namespace: obj.Namespace,
+			Reason:    obj.Reason,
+			Message:   obj.Message,
+			Count:     obj.Count,
+		}
+		fmt.Printf("processed event: %+v\nresulting new interval: %s from: %s to %s\n", summaryObj, condition.Message, pathoFrom, to)
```

Comment:
Suggest not declaring a struct just for the purposes of logging with %v, I'd just format the string as you wish with %s and additional params. I don't know for sure it's any kind of performance hit it just isn't something I've seen done before.